openreview/workflows/process/reviewer_role_process.py

```diff
@@ -0,0 +1,41 @@
+def process(client, invitation):
+
+    print('Compute reviewer roles', invitation.id)
+    domain = client.get_group(invitation.domain)
+    review_name = domain.content.get('review_name', {}).get('value', 'Official_Review')
+    review_invitation_id = f'{domain.id}/-/{review_name}'
+    submission_name = domain.content.get('submission_name', {}).get('value', 'Submission')
+    print('Get reviews')
+    reviews = client.get_notes(parent_invitations=review_invitation_id, stream=True)
+
+    print('Get review signatures')
+    signatures_by_id = { g.id:g for g in client.get_all_groups(prefix=f'{domain.id}/{submission_name}') }
+
+    print('Count reviews by reviewer')
+    review_signatures = [r.signatures[0] for r in reviews]
+
+    reviewers = set()
+    for signature in review_signatures:
+        reviewer_profile = signatures_by_id.get(signature).members[0]
+        reviewers.add(reviewer_profile)
+
+    print('Use profile ids as keys')
+    all_profiles = openreview.tools.get_profiles(client, list(reviewers), as_dict=True)
+
+    
+    tags_by_profile = {}
+
+    for profile_id, profile in all_profiles.items():
+        if profile.id.startswith('~') and profile.id not in tags_by_profile:
+            tags_by_profile[profile.id] = openreview.api.Tag(
+                invitation=invitation.id,
+                signature=domain.id,
+                profile=profile.id
+            )
+    
+    print('Post profile tags', len(tags_by_profile))
```

Comment:
My only concern with this approach is the amount of memory we need to run this process function. Hopefully we won't face any issues. Just something to keep in mind.

Reply:
the concern is loading ALL reviewers profiles?

We already do this in the setup matching to compute conflicts. 

This invitation is meant to run only once BTW.

I'm open to other suggestions.

Reply:
> the concern is loading ALL reviewers profiles?

Yes

> I'm open to other suggestions.

I think it's fine for now. If we see memory issues we could post the tags in batches.